packages/legend-extension-dsl-persistence/src/models/protocols/pure/v1/transformation/pureGraph/to/V1_PersistenceBuilder.ts

```diff
@@ -0,0 +1,519 @@
+  BatchIdTransactionMilestoning,
+  BatchPersister,
+  BitemporalDelta,
+  BitemporalSnapshot,
+  CronTrigger,
+  DateTimeAuditing,
+  DateTimeTransactionMilestoning,
+  DateTimeValidityMilestoning,
+  DeduplicationStrategy,
+  DeleteIndicatorMergeStrategy,
+  EmailNotifyee,
+  FlatTarget,
+  IngestMode,
+  ManualTrigger,
+  MaxVersionDeduplicationStrategy,
+  MergeStrategy,
+  MultiFlatTarget,
+  NoAuditing,
+  NoDeduplicationStrategy,
+  NoDeletesMergeStrategy,
+  NontemporalDelta,
+  NontemporalSnapshot,
+  Notifier,
+  Notifyee,
+  OpaqueTarget,
+  PagerDutyNotifyee,
+  Persister,
+  PropertyAndFlatTarget,
+  SourceSpecifiesFromAndThruDateTime,
+  SourceSpecifiesFromDateTime,
+  StreamingPersister,
+  TargetShape,
+  TransactionMilestoning,
+  TransactionScope,
+  Trigger,
+  UnitemporalDelta,
+  UnitemporalSnapshot,
+  ValidityDerivation,
+  ValidityMilestoning,
+} from '../../../../../../metamodels/pure/model/packageableElements/persistence/Persistence';
+import { getPersistence } from '../../../../../../../graphManager/DSLPersistence_GraphManagerHelper';
+import {
+  GraphBuilderError,
+  IdentifiedConnection,
+  type V1_GraphBuilderContext,
+} from '@finos/legend-graph';
+import { guaranteeNonEmptyString } from '@finos/legend-shared';
+import type { V1_IdentifiedConnection } from '@finos/legend-graph/lib/models/protocols/pure/v1/model/packageableElements/runtime/V1_Runtime';
+import { V1_ProtocolToMetaModelConnectionBuilder } from '@finos/legend-graph/lib/models/protocols/pure/v1/transformation/pureGraph/to/V1_ProtocolToMetaModelConnectionBuilder';
+
+/**********
+ * persistence
+ **********/
+
+export const V1_buildPersistence = (
+  protocol: V1_Persistence,
+  context: V1_GraphBuilderContext,
+): void => {
+  const path = context.graph.buildPath(protocol.package, protocol.name);
+  const persistence = getPersistence(path, context.graph);
+  persistence.documentation = guaranteeNonEmptyString(
+    protocol.documentation,
+    `Persistence 'documentation' field is missing or empty`,
+  );
+  persistence.trigger = V1_buildTrigger(protocol.trigger, context);
+  persistence.service = context.resolveService(protocol.service);
+  persistence.persister = V1_buildPersister(protocol.persister, context);
+  persistence.notifier = V1_buildNotifier(protocol.notifier, context);
+};
+
+/**********
+ * trigger
+ **********/
+
+export const V1_buildTrigger = (
+  protocol: V1_Trigger,
+  context: V1_GraphBuilderContext,
+): Trigger => {
+  if (protocol instanceof V1_CronTrigger) {
+    const trigger = new CronTrigger();
+    trigger.minutes = protocol.minutes;
+    trigger.hours = protocol.hours;
+    trigger.dayOfMonth = protocol.dayOfMonth;
+    trigger.month = protocol.month;
+    trigger.dayOfWeek = protocol.dayOfWeek;
+    return new CronTrigger();
+  } else if (protocol instanceof V1_ManualTrigger) {
+    return new ManualTrigger();
+  }
+  throw new GraphBuilderError(`Unrecognized trigger '${protocol}'`);
```

Comment:
Let's follow the same message pattern. Also, would be nice if we throw `UnsupportedOperationError` instead of `GraphBuidlerError` at these kinds of place.

Reply:
Fixed in #990